### PR TITLE
Revert "Revert "Mjr/xss app manager""

### PR DESCRIPTION
**Pull request by mjriley**
Corrects encoding errors encountered in #29218. In addition, refactored to try to prevent these type of errors in the future. The issue was caused because I moved all mark_safe behavior into the trans method, when this method wasn't always used for HTML output, and would cause issues if it unnecessarily escaped that output. This change tries to prevent future regressions by:

Moving all complicated translation/formatting logic into the _trans helper method. This should make it easier to understand when each top-level filter produces HTML output, and thus needs to be escaped or not.
Documenting the expected output for each filter. Again, this should make it easier to understand when each filter produces HTML output.
No longer escaping the trans or clean_trans filters. These were the filters used outside of an HTML context, and escaping caused issues. Their output still needs to be escaped in an HTML-context, but because it is no longer marked as safe, Django's templates will escape it (assuming auto-escaping is on).
Additionally, conditional escapes are used to prevent double-escaping when the output is already safe.

Probably easiest to review commit-by-commit.

### CODE REVIEW

File: corehq/apps/app_manager/tests/templatetags/test_xforms_extras.py [hunk 1]
```diff
@@ -40,22 +35,6 @@ def test_secondary_language_includes_indicator(self):
...
-    def test_prefix_puts_indicator_in_front_of_translation(self):
-        translation_dict = {'en': 'English Output'}
-        result = trans(translation_dict, prefix=True)
-        self.assertEqual(result, ' [en] English Output')
-
-    def test_strip_tags_removes_tags_from_html_indicator(self):
-        translation_dict = {'en': 'English Output'}
-        result = trans(translation_dict, use_delim=False, strip_tags=True)
-        self.assertEqual(result, 'English Output en ')
-
```
Comment (dannyroberts) [inline corehq/apps/app_manager/tests/templatetags/test_xforms_extras.py]: Why remove the tests? Are these combinations never used anywhere or something?

File: corehq/apps/app_manager/templatetags/xforms_extras.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def _trans(name,
         return pattern.format(affix=affix, name=n)
 
 
-@register.filter
+@register.filter(is_safe=True)
```
Comment (dannyroberts) [inline corehq/apps/app_manager/templatetags/xforms_extras.py:83]: `is_safe` here means "This filter doesn't add any HTML-unsafe characters (<>&, etc) that weren't already in the input", right? I think that's true about this function (that it doesn't add any html tags), but I wanted to check.

**Review by dannyroberts — APPROVED** [review]

File: corehq/apps/app_manager/templatetags/xforms_extras.py (hunk 2 above)
Comment (mjriley) [reply corehq/apps/app_manager/templatetags/xforms_extras.py:83]: Correct. I was going off the documentation [here](https://docs.djangoproject.com/en/3.1/howto/custom-template-tags/#filters-and-auto-escaping), where it seemed like this was a reasonable way to document that this didn't add any unsafe HTML characters.

File: corehq/apps/app_manager/tests/templatetags/test_xforms_extras.py (hunk 1 above)
Comment (mjriley) [reply corehq/apps/app_manager/tests/templatetags/test_xforms_extras.py]: Yeah. Initially I thought `trans` was a general-purpose method that could take any combination of parameters, but then I realized we're only ever using it to support the 4(?) filters that I extracted. Leaving in tests that didn't support real use cases (or were being tested through their respective filters) felt brittle.

Comment (dannyroberts) [reply corehq/apps/app_manager/tests/templatetags/test_xforms_extras.py]: Okay yeah that makes total sense, thanks

**Review by millerdev — APPROVED** [review]

**Review by millerdev — APPROVED** [review]

**Pull request merged**